pkg/cloud/aws/converters/instance.go

```diff
@@ -36,14 +38,18 @@ func SDKToInstance(v *ec2.Instance) *v1alpha1.Instance {
 		EBSOptimized: v.EbsOptimized,
 	}
 
+	// Extract IAM Instance Profile name from ARN
+	split := strings.Split(aws.StringValue(v.IamInstanceProfile.Arn), "instance-profile/")
```

Comment:
We should also check that `v.IamInstanceProfile` isn't nil.

Reply:
done!